screens/QRCodeScreen.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import {
+  StyleSheet,
+  Text,
+  View,
+  Alert,
+} from 'react-native';
+import { BarCodeScanner, Permissions } from 'expo';
+
+const styles = StyleSheet.create({
+  role: {
+    backgroundColor: '#fff',
+    textAlign: 'center',
+    paddingVertical: 10,
+  },
+});
+
+export default class QRCodeScreen extends React.Component {
+  state = {
+    hasCameraPermission: null,
+  }
+
+  async componentDidMount() {
+    const { status } = await Permissions.askAsync(Permissions.CAMERA);
+    this.setState({ hasCameraPermission: status === 'granted' });
```

Comment:
Why do we have to repeat the same `hasCameraPermission` logic both here and in `QRCodeScanner.jsx`?

Reply:
This file should be removed